### ISSUE TITLE
module.networking.aws_default_route_table Error: Provider produced inconsistent result after apply

### ISSUE THREAD

**Issue by mariaiano** [terraform-providers/terraform-provider-aws#9009]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
Terraform v0.12.2
+ provider.aws v2.15.0

### Affected Resource(s)

* aws_default_route_table

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```
https://drive.google.com/open?id=1V0bn0OpZ8jrLKmml6XYUSSk7gQz8bxAN

### Debug Output

https://drive.google.com/open?id=17udS9g1dBo3uI-IhkI2tr9Nbz_9B0UEr

### Expected Behavior

module.networking.aws_default_route_table.dio_private_rt should have been created. 

### Actual Behavior

Instead I got this error:

```
Error: Provider produced inconsistent result after apply

When applying changes to
module.networking.aws_default_route_table.dio_private_rt, provider "aws"
produced an unexpected new value for was present, but now absent.

This is a bug in the provider, which should be reported in the provider's own
issue tracker.
```
### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by mariaiano** [comment]
I wonder why there aren't more people reporting this. Is there some workaround I'm missing?

**Comment by FernandoMiguel** [comment]
@mariaiano i get this all the time :(
`aws_cloudwatch_log_stream.log-stream,
provider "aws" produced an unexpected new value for was present, but now
absent.`

**Comment by tiny-dancer** [comment]
we're seeing this as well, starting to wrap TF in retries.

`
When applying changes to aws_vpc_endpoint_route_table_association.db_s3[1], provider "aws" produced an unexpected new value for was present, but now absent
`

**Comment by hotdawg789** [comment]
Similar response with provider "aws" (hashicorp/aws) 2.25.0.

Error: Provider produced inconsistent result after apply

When applying changes to
aws_vpc_endpoint_route_table_association.pub_rancher_us-east-1a, provider
"aws" produced an unexpected new value for was present, but now absent.

This is a bug in the provider, which should be reported in the provider's own
issue tracker.

**Comment by jayalane** [comment]
I think most people don't use default route table, perhaps because of this.  The extra route tables (one default, one made by terraform) is confusing but functional.

**Comment by jayalane** [comment]
Ah, I have it:  if you specify the VPC this way:

default_route_table_id = aws_vpc.whatever_vpc.id

You get that (which will happen if change aws_route_table to default_route table and then change the key from vpc_id but not the value).  

You want

default_route_table_id = aws_vpc.whatever_vpc.default_route_table.id

**Comment by bflad** [comment]
Hi folks 👋 Sorry you are running into this. Please note I've marked this issue as a bug, but we would prefer to capture each different Terraform resource reporting this error in separate bug reports. Each Terraform resource that hits this error will be due to different code and API behaviors. We are going to treat this report just for the original report of the `aws_default_route_table` resource.

The errant code can be found here: https://github.com/terraform-providers/terraform-provider-aws/blob/101295fdf4b62139d82808d43fd04d84bd1bb435/aws/resource_aws_default_route_table.go#L112-L116

In the `Create` function for the resource, we are returning an empty resource when no default route table is found where we should instead error in this scenario. The error message we can return can be much more helpful to the operator. 👍 

I'll submit the fix for this shortly.

**Comment by bflad** [comment]
The adjustment of the `aws_default_route_table` resource to return a more helpful error in this scenario has been merged and will release with version 2.40.0 of the Terraform AWS Provider this week. If you are seeing the `Provider produced inconsistent result after apply` with other resources (or this resource after this fix is released) please submit a new [bug report](https://github.com/terraform-providers/terraform-provider-aws/issues/new?template=Bug_Report.md) following the issue template for further triage. Thanks.
